src/lib/y2network/dialogs/rename_interface.rb

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm/popup"
+require "y2network/widgets/custom_interface_name"
+require "y2network/widgets/rename_hwinfo"
+
+module Y2Network
+  module Dialogs
+    # This dialog allows the user to rename a network interface
+    #
+    # Is works in a slightly different way depending on the interface.
+    #
+    # * For physical interfaces, it allows the user to select between using
+    #   the MAC adddress or the Bus ID, which are present in the Hwinfo object
+    #   associated to the interface.
+    # * For not connected interfaces ({FakeInterface}), as the hardware is not present,
+    #   the user must specify the MAC or the Bus ID by hand (NOT IMPLEMENTED YET).
+    # * For virtual interfaces, like bridges, only the name can be chaned (no hardware
+    #   info at all) (NOT IMPLEMENTED YET).
+    class RenameInterface < CWM::Popup
+      def initialize(builder)
+        textdomain "network"
+
+        @builder = builder
+        @old_name = interface.name
+      end
+
+      # Runs the dialog
+      def run
+        ret = super
+        return unless ret == :ok
+        renaming_mechanism = rename_hwinfo_widget.value unless virtual_interface?
+        @builder.rename_interface(name_widget.value, renaming_mechanism)
+        name_widget.value
```

Comment:
if it is intention to return new name which is different to common run method, please document it. Also document that it returns nil if not confirmed. And last but not least, please also document that modify builder.

Reply:
I just tried to keep the same behaviour than [EditNicName](https://github.com/yast/yast-network/blob/master/src/lib/network/edit_nic_name.rb#L84). Regarding modifying the builder, I could move the change to the calling widget if I return the renaming mechanism and the name. Is it ok for you?

Reply:
Refactored and simplified.